enforce.go

```diff
@@ -149,16 +149,8 @@ func (e *enforcer) isReturnedWithNonNilError(stack []ast.Node) bool {
 	}
 
 	// Find the nearest function's type for the return statement.
-	var ftype *ast.FuncType
-	for idx := retIdx - 1; idx >= 0; idx-- {
-		switch n := stack[idx].(type) {
-		case *ast.FuncDecl:
-			ftype = n.Type
-		case *ast.FuncLit:
-			ftype = n.Type
-		}
-	}
-	if ftype == nil {
+	ftype, ok := parentFuncType(stack[:retIdx])
+	if !ok || ftype == nil {
```

Comment:
to check my understanding, it should be impossible to hit the nil pointer panic on line 159 because `ftype.Results` should always exist now? 

To be consistent with this impossible check `ftype == nil`, I wonder if we should still guard against it? The one downside to not panicing, is you mask this kind of bug.

Reply:
I think it should be impossible since this code should only run when returning a struct with required fields, but that requires the function to have non-empty returns (hence `ftype.Results != nil`). That said, the AST would technically allow that (but the type checker would disallow it).

We only ran into that case because the function signature matching was incorrect, but I do think it's worth guarding against a panic in that case, since the code already handles `if len(returnTypes) == 0 {`.